lib/membrane_rtmp_plugin/rtmp_server.ex

```diff
@@ -1,6 +1,12 @@
 defmodule Membrane.RTMP.Server do
   @moduledoc """
   A simple RTMP server, which handles each new incoming connection.
+
+  When new client connects to the server, it goes into :client_waiting_queue and its RTMP handshake will remanin unfinished.
+  Only when pipeline tries to pull data from client, its handshake will be finished, and client will be registered.
```

Comment:
I wouldn't mention the "pipeline" - the `RTMPServer` is meant to be used outside of Membrane pipeline as well and using it with the `Membrane.RTMP.Source` is just one of the options.